### PR TITLE
Encapsulating element tag parsing

### PR DESCRIPTION
**Pull request by DenverM80**
This pull request includes changes from pull request https://github.com/SpectraLogic/ds3_autogen/pull/228 that was opened 6 days ago, plus fixes issues related to parsing of elements with and without encapsulating element tags.

### CODE REVIEW

**Comment by rpmoore** [comment]
This is now out of date.

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/StructConverter.java [hunk 1]
```diff
@@ -93,4 +94,11 @@ public static boolean hasComplexArrayMembers(final ImmutableList<StructMember> s
         return false;
     }
 
+    private static boolean hasWrapper(final Ds3Element ds3Element) {
+        System.out.println(ds3Element.getName());
```
Comment (RachelTucker) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/StructConverter.java]: println should be changed to a logging message

**Comment by DenverM80** [comment]
updated.

File: ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/StructConverter.java (hunk 1 above)
Comment (DenverM80) [inline ds3-autogen-c/src/main/java/com/spectralogic/ds3autogen/c/converters/StructConverter.java]: I'll remove it completely

**Pull request merged**